From b0b2fc815e514221f01384f39fbfbff65d897e1c Mon Sep 17 00:00:00 2001
From: Remington Brasga <rbrasga@uci.edu>
Date: Wed, 10 Jul 2024 00:12:44 +0000
Subject: [PATCH] jfs: UBSAN: shift-out-of-bounds in dbFindBits
Git-commit: b0b2fc815e514221f01384f39fbfbff65d897e1c
Patch-mainline: v6.12-rc1
References: git-fixes

Fix issue with UBSAN throwing shift-out-of-bounds warning.

Reported-by: syzbot+e38d703eeb410b17b473@syzkaller.appspotmail.com
Signed-off-by: Remington Brasga <rbrasga@uci.edu>
Signed-off-by: Dave Kleikamp <dave.kleikamp@oracle.com>
Acked-by: Anthony Iliopoulos <ailiop@suse.com>

---
 fs/jfs/jfs_dmap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/jfs/jfs_dmap.c b/fs/jfs/jfs_dmap.c
index 5713994328cb..ccdfa38d7a68 100644
--- a/fs/jfs/jfs_dmap.c
+++ b/fs/jfs/jfs_dmap.c
@@ -3022,7 +3022,7 @@ static int dbFindBits(u32 word, int l2nb)
 
 	/* scan the word for nb free bits at nb alignments.
 	 */
-	for (bitno = 0; mask != 0; bitno += nb, mask >>= nb) {
+	for (bitno = 0; mask != 0; bitno += nb, mask = (mask >> nb)) {
 		if ((mask & word) == mask)
 			break;
 	}
-- 
2.47.0

